aspnetcore/blazor/components/css-isolation.md

```diff
@@ -0,0 +1,165 @@
+
+The following example shows a parent component called `Parent` with a child component called `Child`.
+
+`Parent.razor`:
+
+```razor
+@page "/parent"
+
+<div>
+    <h1>Parent component</h1>
+
+    <Child />
+</div>
+```
+
+`Child.razor`:
+
+```razor
+<h1>Child Component</h1>
+```
+
+Update the `h1` declaration in `Parent.razor.css` with the `::deep` combinator to signify the `h1` style declaration must apply to the parent component and its children:
+
+```css
+::deep h1 { 
+    color: red;
+}
+```
+
+The `h1` style now applies to the `Parent` and `Child` components without the need to create a separate scoped CSS file for the child component.
+
+> [!NOTE]
+> The `::deep` combinator only works with descendant elements. The following HTML structure applies the `h1` styles to components as expected:
+> 
+> ```razor
+> <div>
+>     <h1>Parent</h1>
+>
+>     <Child />
+> </div>
+> ```
+>
+> In this scenario, ASP.NET Core applies the parent component's scope identifier to the `div` element, so the browser knows to inherit styles from the parent component.
+>
+> However, excluding the `div` element removes the descendant relationship, and the style is **not** applied to the child component:
+>
+> ```razor
+> <h1>Parent</h1>
+>
+> <Child />
+> ```
+
+## CSS preprocessor support
+
+CSS preprocessors are useful for improving CSS development by utilizing features such as variables, nesting, modules, mixins, and inheritance. While CSS isolation doesn't natively support CSS preprocessors such as Sass or Less, integrating CSS preprocessors is seamless as long as preprocessor compilation occurs before Blazor rewrites the CSS selectors during the build process. Using Visual Studio for example, configure existing preprocessor compilation as a **Before Build** task in the Visual Studio Task Runner Explorer.
+
+Many third-party NuGet packages, such as [Delegate.SassBuilder](https://www.nuget.org/packages/Delegate.SassBuilder), can compile SASS/SCSS files at the beginning of the build process before CSS isolation occurs, and no additional additional configuration is required.
+
+## CSS isolation configuration
+
+CSS isolation is designed to work out-of-the-box but provides configuration for some advanced scenarios, such as when there are dependencies on existing tools or workflows.
+
+### Customize scope identifier format
+
+By default, scope identifiers use the format `b-<10-character-string>`. To customize the scope identifier format, update the project file to a desired pattern:
+
+```xml
+<ItemGroup>
+    <None Update="MyComponent.razor.css" CssScope="my-custom-scope-identifier" />
+</ItemGroup>
+```
+
+In the preceding example, the CSS generated for `MyComponent.Razor.css` changes its scope identifier from `b-<10-character-string>` to `my-custom-scope-identifier`.
+
+### Change base path for static web assets
```

Comment:
Is this the right place to include this heading?

Reply:
Yeah, there's no natural way unless we include it in the main section where it's discussed. I personally wanted to save the config to the end as it is optional and probably rarely used.